### PR TITLE
Add support for INT64 types in TensorRT constant layer calibration

### PR DESCRIPTION
**Pull request by kevinch-nv**
### Description
- TensorRT supports INT64 types since TRT 10, so add support for it when calibrating constant layers.
- Add more logging statements for helpful debug messages to users when there is an issue reading from a calibration cache.

### Motivation and Context
- This change is necessary for functional parity with TRT 10

### CODE REVIEW

**Comment by chilo-ms** [comment]
Please help fix Lint / Python format

**Review by chilo-ms — APPROVED** [review]

File: onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc [hunk 1]
```diff
@@ -108,13 +111,19 @@ bool SetDynamicRange(nvinfer1::INetworkDefinition& network, std::unordered_map<s
             case nvinfer1::DataType::kINT32:
               weight = static_cast<const int32_t*>(trt_weights.values)[k];
               break;
+#if NV_TENSORRT_MAJOR >= 10
+            case nvinfer1::DataType::kINT64:
+              weight = static_cast<const int64_t*>(trt_weights.values)[k];
```
Comment (chilo-ms) [inline onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc:116]: Converting "const int64_t" to "double" may lost data.
````
warning C4244: '=': conversion from 'const int64_t' to 'double', possible loss of data
````

Comment (jywu-msft) [reply onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc:116]: @kevinch-nv, fyi this is causing the build to fail.

Comment (kevinch-nv) [reply onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc]: Updated

**Comment by chilo-ms** [comment]
Please help fix the format.

BTW, i create a duplicated PR in case this PR can't be updated for patch release timeline.

**Comment by jywu-msft** [comment]
> Please help fix the format.
> 
> BTW, i create a duplicated PR in case this PR can't be updated for patch release timeline.

https://github.com/microsoft/onnxruntime/pull/21101

**Comment by jywu-msft** [comment]
superseded by https://github.com/microsoft/onnxruntime/pull/21101

**Pull request closed without merging**